lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -272,10 +274,10 @@ class QcFlagService {
      * Delete single instance of QC flag
      * @param {number} id QC flag id
      * @param {object} [relations] QC Flag entity relations
-     * @param {UserIdentifier+IsAdminOption} [relations.user] user
+     * @param {UserWithRoles} [relations.user] user
      * @return {Promise<QcFlag>} promise
      */
-    async delete(id, relations) {
+    async delete(id, relations = {}) {
```

Comment:
Why allowing empty?

Reply:
I think we are mixing things here.
In the service layer, you can expect the expected parameters to be provided in. The errors you handle here might be that the user with given id is not found, but you can expect an id to be provided.
It's the role of the API to be sure that user provided an ID.

Reply:
The check in `getUser` is done because js do not allow us to expect at least one of the two and because of the partial, I prefer to check. Here you can clearly define in your signature that you expect the relation, and calling code must comply.

Reply:
made mandatory